wireapp/wire-android
# fix: audio message filter toggle (WPB-6406)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6406" title="WPB-6406" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-6406</a>  [Android] Apply Filters to Audio Recording
  </td></table>
  <br />
 




----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Wrong handling of checkbox to apply filters

### Causes (Optional)

Misunderstood of ACs caught only after design review

### Solutions

- Checkbox on last state (ready to send audio file) is now enabled.
- Checkbox is also enabled whilst playing the preview audio, which will also change the preview audio played.
- Moved `AudioEffect.applyEffectM4A(..)` to a separate use case in order to fully test the view model (as we cannot mock/test this class)

### Testing

#### Test Coverage (Optional)


#### How to Test

- Record Audio (can check or not the apply filters options in the beginning)
- on the last state (ready to send audio), can listen to recorded audio in original form (no selected checkbox) or with effects (selected checkbox)

Note: audio will continue from last position when selecting/unselecting the checkbox (meaning if audio is 50% listen and checkbox is selected, then it will continue from 50% of it)
